src/Symfony/Component/Messenger/Tests/Transport/AmqpExt/ConnectionTest.php

```diff
@@ -255,6 +255,36 @@ public function testPublishWithQueueOptions()
         $connection = Connection::fromDsn('amqp://localhost/%2f/messages?queue[attributes][delivery_mode]=2&queue[attributes][headers][token]=uuid&queue[flags]=1', [], $factory);
         $connection->publish('body', $headers);
     }
+
+    public function testItCanPublishWithTheDefaultQueueRoutingKey()
+    {
+        $factory = new TestAmqpFactory(
+            $amqpConnection = $this->getMockBuilder(\AMQPConnection::class)->disableOriginalConstructor()->getMock(),
+            $amqpChannel = $this->getMockBuilder(\AMQPChannel::class)->disableOriginalConstructor()->getMock(),
+            $amqpQueue = $this->getMockBuilder(\AMQPQueue::class)->disableOriginalConstructor()->getMock(),
+            $amqpExchange = $this->getMockBuilder(\AMQPExchange::class)->disableOriginalConstructor()->getMock()
+        );
+
+        $amqpExchange->expects($this->once())->method('publish')->with('body', 'my_key');
+
+        $connection = Connection::fromDsn('amqp://localhost/%2f/messages?queue[routing_key]=my_key', [], $factory);
```

Comment:
Shouldn't it be in `exchange[routing_key]` actually? Because it's when we publish to the exchange.

Reply:
Ya, sounds good to me. Working on it. Ty.

Reply:
Let's move them to `exchangeConfiguration`. But let's add a BC-layer (with a deprecation warning) for people using the `queueConfiguration`, because it won't really fail their code it might be hard for people to know this has changed.

Reply:
Ok, working on it.

Reply:
With my last commit, people using a queue routing key have to define an exchange routing key to prevent the warning. Both keys must be identicals to reproduce the same comportment in their applications.

It sounds OK to me but I'm not confortable with the complexity and the extra configuration it introduces. Are you ?